tools/src/bin/collect.rs

```diff
@@ -1142,4 +1318,148 @@ mod tests {
+        let hpke_keypair = generate_test_hpke_config_and_private_key();
+        let encoded_hpke_config = URL_SAFE_NO_PAD.encode(hpke_keypair.config().get_encoded());
+        let encoded_private_key = URL_SAFE_NO_PAD.encode(hpke_keypair.private_key().as_ref());
+
+        let task_id = random();
+        let leader = Url::parse("https://example.com/dap/").unwrap();
+        let auth_token = AuthenticationToken::DapAuth(random());
+
+        let mut expected = Options {
+            subcommand: Some(Subcommands::NewJob {
+                collection_job_id: None,
+            }),
+            task_id,
+            leader: leader.clone(),
+            authentication: AuthenticationOptions {
+                dap_auth_token: Some(auth_token.clone()),
+                authorization_bearer_token: None,
+            },
+            hpke_config: Some(hpke_keypair.config().clone()),
+            hpke_private_key: Some(hpke_keypair.private_key().clone()),
+            hpke_config_json: None,
+            vdaf: VdafType::Count,
+            length: None,
+            bits: None,
+            query: QueryOptions {
+                batch_interval_start: Some(1_000_000),
+                batch_interval_duration: Some(1_000),
+                batch_id: None,
+                current_batch: false,
+            },
+        };
+        let task_id_encoded = URL_SAFE_NO_PAD.encode(task_id.get_encoded());
+        let correct_arguments = [
+            "collect",
+            &format!("--task-id={task_id_encoded}"),
+            "--leader",
+            leader.as_str(),
+            &format!("--dap-auth-token={}", auth_token.as_str()),
+            &format!("--hpke-config={encoded_hpke_config}"),
+            &format!("--hpke-private-key={encoded_private_key}"),
+            "--vdaf",
+            "count",
+            "--batch-interval-start",
+            "1000000",
+            "--batch-interval-duration",
+            "1000",
+            "new-job",
+        ];
+        match Options::try_parse_from(correct_arguments) {
+            Ok(got) => assert_eq!(got, expected),
+            Err(e) => panic!("{}\narguments were {:?}", e, correct_arguments),
+        }
+
+        let collection_job_id = random();
+        expected.subcommand = Some(Subcommands::NewJob {
+            collection_job_id: Some(collection_job_id),
+        });
+        let correct_arguments = [
+            "collect",
+            &format!("--task-id={task_id_encoded}"),
+            "--leader",
+            leader.as_str(),
+            &format!("--dap-auth-token={}", auth_token.as_str()),
+            &format!("--hpke-config={encoded_hpke_config}"),
+            &format!("--hpke-private-key={encoded_private_key}"),
+            "--vdaf",
+            "count",
+            "--batch-interval-start",
+            "1000000",
+            "--batch-interval-duration",
+            "1000",
+            "new-job",
+            &format!("{collection_job_id}"),
```

Comment:
If the random collection job ID starts with a hyphen when encoded, this test would flake. Since this is a positional argument, we can't use the --flag=value workaround, as above, but we can use `--` to indicate that all subsequent arguments are positional arguments, not flags. (I haven't tested this myself, but hopefully one of these works)
```suggestion
            "--",
            "new-job",
            &format!("{collection_job_id}"),
```

```suggestion
            "new-job",
            "--",
            &format!("{collection_job_id}"),
```

Reply:
Nice catch! `new-job -- collection_job_id` works.